### PR TITLE
Class 06

### PR DESCRIPTION
**Pull request by derekgraham**
Completed main goals and stretch goals. 

Data file format changed to .json. Loading data from .json source and saving to local storage. At load time we check the digest of the header to see if anything has changed. If so, we dump local storage and load remote and save it, then display it. If it hasn't changed, we load from local and store it.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,67 +1,116 @@
...
+                    created
+                    <time title="{{publishedDate}}" pubdate="{{publishedDate}}"" datetime="2000-01-01">{{relativePubDate}}</time>
+                </div>
+            </header>
+            <section class="project-body">{{description}}</section>
+            {{#if media.imageURL}}
+            <section class="project-image">
+                <a class='project-image-link' href=""><img class='project-image' src="{{media.imageURL}}"></a>
+            </section>
+            {{/if}} {{#if media.video}}
```
Comment (amyeleek) [inline index.html:30]: You can do else in Handlebars, unless you want to have an image and a video at the same time.

File: scripts/portfolio.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    } else {
+      out += months[myMonths < 0 ? (12-myMonths) % 12 : Math.abs(12-myMonths % 12 - 1)] + ' months ago';
+    }
+  } else out += parseInt(myDays / 365) + ' years ago';
+  return out;
+};
+
+Project.prototype.generateHtml = function (){
+
+  Handlebars.registerHelper('if', function(conditional, options) {
```
Comment (amyeleek) [inline scripts/portfolio.js:26]: What's this? You get if for free in Handlebars.

File: scripts/projectView.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+var portfolioPage = {};
+
+portfolioPage.populateFilter = function(template_id) {
+  var $source = $(template_id).html();
+  var filter = Handlebars.compile($source);
+  return filter({myProjects: projects});
+};
+
+
+Handlebars.registerHelper('list', function(items, options) {
```
Comment (amyeleek) [inline scripts/projectView.js:11]: I don't think this is working how you want it to and I'm not actually sure what you're trying to do here.

File: scripts/projectView.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    $('.main-nav .tab:first').click();
+
+  });
+
+
+  $('.main-nav .tab:first').click();
+};
+
+
+//fixme: create some longer bogus data and get show more / less working
```
Comment (amyeleek) [inline scripts/projectView.js:62]: http://www.lipsum.com/

File: scripts/projectView.js (hunk 3 above)
Comment (derekgraham) [inline scripts/projectView.js:11]: That is correct. It broke when I moved from 'projects' as an array to Portfolio.all as the array holder.  The 'list' block helper creates the category dropdown elements.

**Pull request merged**